cmd/k8s-operator/operator.go

```diff
@@ -222,11 +229,44 @@ waitOnline:
 	}
 
 	startlog.Infof("Startup complete, operator running")
+	if shouldRunAuthProxy == "true" {
+		authProxyListener, err := s.Listen("tcp", ":443")
+		if err != nil {
+			startlog.Fatalf("could not listen on :443: %v", err)
+		}
+		go runAuthProxy(lc, authProxyListener)
+	}
 	if err := mgr.Start(signals.SetupSignalHandler()); err != nil {
 		startlog.Fatalf("could not start manager: %v", err)
 	}
 }
 
+func runAuthProxy(lc *tailscale.LocalClient, ls net.Listener) {
+	u, err := url.Parse(fmt.Sprintf("https://%s:%s", os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT_HTTPS")))
+	if err != nil {
+		panic(err)
+	}
+	kc, err := kube.New()
+	if err != nil {
+		panic(err)
+	}
+	p := &httputil.ReverseProxy{
+		Director: func(r *http.Request) {
+			who, err := lc.WhoIs(r.Context(), r.RemoteAddr)
+			if err != nil {
+				panic(err)
```

Comment:
Wrap the reverse proxy in another handler that does the WhoIs lookup and passes it in request context? That way if the whois fails you can respond with a 403 or 500 before ReverseProxy is invoked instead of panicking.

Reply:
Good idea, done.